Plugins/chronoemulator/src/story.h

```diff
@@ -0,0 +1,51 @@
+#ifndef CHRONOEMULATOR_STORY_H
+#define CHRONOEMULATOR_STORY_H
+
+#include "vector"
+
+namespace chronoemulator
+{
+/**
+ * @brief Represents a single story with a series of timestamped events.
+ *
+ * This class provides functionalities to record events happening at specific timestamps
+ * and to replay events that have been recorded up until a specific timestamp.
+ */
+class StoryHandle
+{
+public:
+    std::string storyName; ///< The name of the story.
+    std::vector <std::pair <uint64_t, std::string>> events; ///< A list of events and their timestamps.
```

Comment:
std::vector is not a best choice for the events container. I imagine most of hte emulation would revolve around lookup and retreaval of events based on the timestamp as a key. Key based data structure would be more efficient here